src/MessagePack.UnityClient/Assets/Scripts/MessagePack/Resolvers/CompositeResolver.cs

```diff
@@ -38,124 +30,102 @@ public CompositeResolver()
         /// The composite resolver that includes the AOT code generated resolver and several standard ones
         /// that the AOT code generator expects to be present.
         /// </returns>
-        public static CompositeResolver CreateForAot(IFormatterResolver aotResolver)
+        public static IFormatterResolver CreateForAot(IFormatterResolver aotResolver)
         {
-            var composite = new CompositeResolver();
-            composite.RegisterResolver(aotResolver);
-
-            composite.RegisterResolver(
-                BuiltinResolver.Instance,
-                AttributeFormatterResolver.Instance,
-                PrimitiveObjectResolver.Instance);
+            if (aotResolver is null)
+            {
+                throw new ArgumentNullException(nameof(aotResolver));
+            }
 
-            return composite;
+            return new CachingResolver(
+                EmptyFormattersByType,
+                new ReadOnlyCollection<IFormatterResolver>(
+                    new[]
+                    {
+                        aotResolver,
+                        BuiltinResolver.Instance,
+                        AttributeFormatterResolver.Instance,
+                        PrimitiveObjectResolver.Instance,
+                    }));
         }
 
-        /// <summary>
-        /// Adds a resolver to this composite resolver.
-        /// </summary>
-        /// <param name="resolver">The resolver to add.</param>
-        /// <remarks>
-        /// Each registered resolver is appended to the end of the list of resolvers to try for each requested type.
-        /// Registered formatters take precedence over all resolvers.
-        /// </remarks>
-        public void RegisterResolver(IFormatterResolver resolver)
+        public static IFormatterResolver Create(IReadOnlyList<object> formatters, IReadOnlyList<IFormatterResolver> resolvers)
```

Comment:
Original CompositeResolver has `params IFormatterResolver[] resolvers` and others.
These overloads are useful in many scenarios.

Reply:
Added